tests/transforms/test_optimization/test_optimization_utils.py

```diff
@@ -56,78 +52,199 @@ def test_find_next_gate(self, wires, op_list, next_gate_idx):
+        (
+            [[[-0.9, 1.2, 0.6]] * 2] * 4,
+            [[[0.4, 0.1, 0.0], [0.7, 0.2, 0.1]]] * 4,
+        ),  # ((4, 2), (4, 2))
+    ]
+
+    @pytest.mark.parametrize("angles_1, angles_2", mixed_batched_angles)
+    def test_full_rot_fusion_mixed_batching(self, angles_1, angles_2):
+        """Test that the fusion of two Rot gates has the same effect as
+        applying the Rots sequentially when the input angles are batched
+        with mixed batching shapes."""
+
+        fused_angles = fuse_rot_angles(angles_1, angles_2)
+
+        reshaped_angles_1 = np.reshape(angles_1, (-1, 3) if np.ndim(angles_1) > 1 else (3,))
+        reshaped_angles_2 = np.reshape(angles_2, (-1, 3) if np.ndim(angles_2) > 1 else (3,))
+        matrix_expected = self.mat_from_prod(reshaped_angles_1.T, reshaped_angles_2.T)
+        matrix_obtained = qml.Rot(*fused_angles.reshape((-1, 3)).T, wires=0).matrix()
+
+        assert qml.math.allclose(matrix_expected, matrix_obtained)
 
     @pytest.mark.autograd
-    @pytest.mark.parametrize(
-        ("angles_1", "angles_2"),
-        [
-            ([0.15, 0.25, -0.90], [-0.5, 0.25, 1.3]),
-            ([0.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
-            ([0.1, 0.2, 0.3], [0.0, 0.0, 0.0]),
-            ([0.0, 0.0, 0.0], [0.2, 0.4, -0.5]),
-            ([0.15, 0.25, -0.90], [-0.15, -0.25, 0.9]),
-            ([0.05, -1.34, 4.12], [-0.8, 0.2, 0.12]),
-            ([0.05, 0.0, 4.12], [-0.8, 0.2, 0.12]),
-            ([0.05, -1.34, 0.0], [-0.8, 0.2, 0.12]),
-            ([0.05, 0.0, 0.1], [-0.2, 0.0, 0.12]),
-            ([0.05, 0.0, 0.0], [0.0, 0.0, 0.12]),
-            ([0.05, 0.0, 0.0], [0.0, 0.0, -0.05]),
-            ([0.05, 0.0, 0.0], [-0.05, 0.0, 0.0]),
-            ([0.0, 0.0, 0.05], [-0.05, 0.0, 0.0]),
-            ([0.0, 0.0, 0.05], [0.0, 0.0, -0.05]),
-            ([0.05, 0.2, 0.0], [0.0, -0.6, 0.12]),
-            ([0.05, -1.34, 4.12], [0.0, 0.2, 0.12]),
-            ([0.05, -1.34, 4.12], [0.3, 0.0, 0.12]),
-            ([np.pi, np.pi / 2, 0.0], [0.0, -np.pi / 2, 0.0]),
-            ([0.9, np.pi / 2, 0.0], [0.0, -np.pi / 2, 0.0]),
-            ([0.9, np.pi / 2, np.pi / 2], [-np.pi / 2, -np.pi / 2, 0.0]),
-        ],
-    )
+    @pytest.mark.parametrize("angles_1, angles_2", generic_test_angles)
     def test_full_rot_fusion_autograd(self, angles_1, angles_2):
         """Test that the fusion of two Rot gates has the same effect as
-        applying the Rots sequentially."""
+        applying the Rots sequentially, in Autograd."""
 
-        def original_ops():
-            qml.Rot(*angles_1, wires=0)
-            qml.Rot(*angles_2, wires=0)
+        angles_1, angles_2 = qml.numpy.array(angles_1), qml.numpy.array(angles_1)
+        matrix_expected = self.mat_from_prod(angles_1, angles_2)
+        fused_angles = fuse_rot_angles(angles_1, angles_2)
+        matrix_obtained = qml.Rot(*fused_angles, wires=0).matrix()
+
+        assert qml.math.allclose(matrix_expected, matrix_obtained)
```

Comment:
Can we bundle these lines into a function to avoid duplication below?

Reply:
Done, worked very well!